Week1/homework/app.js

```diff
@@ -1,11 +1,157 @@
+    author: "Anna Frank",
+  },
+    "meisje_in_de_trein": {
+      title: "Meisje in de trein",
+      language: "Dutch",
+      author: "Paula Hawkins",
+    },
+    "hanna's_verjaardag": {
+      title: "Hanna's verjaardag",
+      language: "Dutch",
+      author: "Gull Akerblom",
+    },
+    "Een_nieuwe_toekomst": {
+      title: "Een nieuwe toekomst",
+      language: "Dutch",
+      author: "Conor Grennan",
+    },
+    "de_geur_van_regen": {
+      title: "De geur van regen",
+      language: "Dutch",
+      author: "Lieneke Dijkzeul",
+    },
+    "detective_orhan": {
+      title: "Detective Orhan",
+      language: "Dutch",
+      author: "Sadik Yemni",
+    },
+    "last_minute": {
+      title: "Last Minute",
+      language: "Dutch",
+      author: "Linda van Rijn",
+    },
+    "de_man_van_de_bloemen": {
+      title: "De man van de bloemen",
+      language: "Dutch",
+      author: "Johan van Caeneghem",
+    },
+    "ren_amina_ren": {
+      title: "Ren Amina ren",
+      language: "Dutch",
+      author: "Annelie Drewsen",
+    },
+    "het_diner": {
+      title: "Het diner",
+      language: "Dutch",
+      author: "Herman Koch",
+    }
+  
+  }
+  //console.log(bookDetails);
+  
+  const bookImg = {
+    anna_frank_haar_leven : "images/Anna Frank.jpeg",
+    meisje_in_de_trein : "images/meisje in de trein.jpg",
+    hannas_verjaardag : "images/Hannas' verjaardag.jpg",
+    Een_nieuwe_toekomst : "images/nieuwe toekomst.jpg",
+    de_geur_van_regen : "images/de geur.jpg",
+    detective_orhan : "images/detective orhan.jpg",
+    last_minute : "images/last minute.jpg",
+    de_man_van_de_bloemen : "images/man van de bloemen.jpg",
+    ren_amina_ren : "images/ren amina.jpg",
+    het_diner : "images/het diner.jpg"
+    };
+ 
+  function bookList () { 
+
+    let bookcontainer = document.getElementById("mybooks");
+    let div = document.createElement("div");
+    bookcontainer.appendChild(div);
+    let li = document.createElement("li");
+    div.appendChild(li);
+    
+    
+
+
+    for (let i = 0; i< bookTitles.length; i++) {
+
+      // let content = document.createTextNode(bookTitles[i]);
+      // li.appendChild(content);
+      let titleList = document.createElement("li")
+      li.appendChild(titleList);
+      let titleContent = document.createTextNode(bookDetails[bookTitles[i]].title);
+      titleList.appendChild(titleContent);
+      let languageList = document.createElement("li")
+      li.appendChild(languageList);
+      let languageContent = document.createTextNode("Language: " + bookDetails[bookTitles[i]].language);
+      languageList.appendChild(languageContent);
+      let authorList = document.createElement("li")
+      li.appendChild(authorList);
+      let authorContent = document.createTextNode("Author: "+ bookDetails[bookTitles[i]].author);
+      authorList.appendChild(authorContent);
+      let img = document.createElement("img");
+      li.appendChild(img);
+      img.src = Object.values(bookImg)[i];
```

Comment:
Huh, I'm surprised to see this, after you were approaching the data access above correctly.

Using `Object.values(bookImg)[i]` hoping to get the same entry is incorrect.  If `bookImg` had been an array of images, in the same order as the `bookTitles` you would have been correct, and would have referred to the correct image like this:
```
img.src = bookImg[i];
```
Objects are a key/value store, and you access the items in the object by referring to the key.  So, if I want to access 
```
"images/Hannas' verjaardag.jpg"
```
I have to get it by asking the object for the value of the `hannas_verjaardag` key.
When I call `Object.values` it gives me back all the items in an array, but the order of the items in the array is not guaranteed.
So, you call it hoping that it will give you back the values in the order you have defined them, but that is not something you can rely on.  As a result, it is very possible that the image displayed for your book is the wrong one. 
You should have accessed the image in the same way you accessed the details. i.e.
```
img.src = bookImg[bookTitles[i]];
```
That would have guaranteed the correct image for each book.